### PR TITLE
[Review Needed]Pytest Style coordinates/test_lammps.py

### PR DESCRIPTION
**Pull request by utkbansal**
Fixes #

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: testsuite/MDAnalysisTests/coordinates/test_lammps.py [hunk 1]
```diff
@@ -27,77 +27,71 @@
...
+    def test_singleframe(self, u):
         with pytest.raises(StopIteration):
-            self.u.trajectory.next()
+            u.trajectory.next()
 
-    def test_seek(self):
+    def test_seek(self, u):
         with pytest.raises(IndexError):
-            self.u.trajectory.__getitem__(1)
+            u.trajectory.__getitem__(1)
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/coordinates/test_lammps.py]: `u.trajectory[1]`

File: testsuite/MDAnalysisTests/coordinates/test_lammps.py [hunk 2]
```diff
@@ -189,177 +183,169 @@ def test_Writer_atoms(self, attr, LAMMPSDATA_partial):
...
-            assert_(W.flavor, self.flavor)
-
-    def test_Writer(self, n_frames=3):
-        W = mda.Writer(self.outfile, n_atoms=self.u.atoms.n_atoms,
+    @pytest.fixture(scope='class')
+    def u(self):
+        return mda.Universe(self.topology, self.trajectory, format=self.format)
+
+    def test_Writer_is_LAMMPS(self, u, tmpdir):
+        outfile = str(tmpdir) + 'lammps-writer-test' + \
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/coordinates/test_lammps.py]: `outfile = os.path.join(str(tmpdir), 'lammps-writer-test' + ext)`

Don't join paths manually!

File: testsuite/MDAnalysisTests/coordinates/test_lammps.py [hunk 3]
```diff
@@ -189,177 +183,169 @@ def test_Writer_atoms(self, attr, LAMMPSDATA_partial):
...
+                  os.path.splitext(self.trajectory)[1]
+        with mda.Writer(outfile, n_atoms=u.atoms.n_atoms,
+                        format=self.format) as W:
+            assert W.flavor, self.flavor
+
+    def test_Writer(self, u, tmpdir, n_frames=3):
+        outfile = str(tmpdir) + 'lammps-writer-test' + \
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/coordinates/test_lammps.py]: see comment above

File: testsuite/MDAnalysisTests/coordinates/test_lammps.py [hunk 4]
```diff
@@ -189,177 +183,169 @@ def test_Writer_atoms(self, attr, LAMMPSDATA_partial):
...
-                            6, err_msg="coordinate mismatch between corresponding frames")
+                            u.trajectory[n_frames - 1].positions,
+                            6,
+                            err_msg="coordinate mismatch between corresponding frames")
 
-    def test_OtherWriter_is_LAMMPS(self):
-        with self.u.trajectory.OtherWriter(self.outfile) as W:
-            assert_(W.flavor, self.flavor)
+    def test_OtherWriter_is_LAMMPS(self, u, tmpdir):
+        outfile = str(tmpdir) + 'lammps-writer-test' + \
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/coordinates/test_lammps.py]: see comment above

File: testsuite/MDAnalysisTests/coordinates/test_lammps.py [hunk 5]
```diff
@@ -189,177 +183,169 @@ def test_Writer_atoms(self, attr, LAMMPSDATA_partial):
...
+                  os.path.splitext(self.trajectory)[1]
+        with u.trajectory.OtherWriter(outfile) as W:
+            assert W.flavor, self.flavor
 
-    def test_OtherWriter(self):
+    def test_OtherWriter(self, u, tmpdir):
         times = []
-        with self.u.trajectory.OtherWriter(self.outfile) as w:
-            for ts in self.u.trajectory[::-1]:
+        outfile = str(tmpdir) + 'lammps-writer-test' + \
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/coordinates/test_lammps.py]: see comment above

File: testsuite/MDAnalysisTests/coordinates/test_lammps.py (hunk 2 above)
Comment (kain88-de) [reply testsuite/MDAnalysisTests/coordinates/test_lammps.py]: In general those manual line breaks with `\` are also not that nice and should be avoided if possible.

**Comment by utkbansal** [comment]
@kain88-de Anything else needed here?

**Pull request merged**